platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/heartbeat/HeartbeatProtocolTests.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.heartbeat;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+
+import com.swirlds.common.system.NodeId;
+import com.swirlds.common.test.fixtures.FakeTime;
+import com.swirlds.platform.Connection;
+import com.swirlds.platform.heartbeats.HeartbeatProtocol;
+import com.swirlds.platform.network.ByteConstants;
+import com.swirlds.platform.network.NetworkMetrics;
+import com.swirlds.platform.network.NetworkProtocolException;
+import com.swirlds.platform.sync.SyncInputStream;
+import com.swirlds.platform.sync.SyncOutputStream;
+import java.io.IOException;
+import java.time.Duration;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+/**
+ * Tests for {@link HeartbeatProtocol}
+ */
+@DisplayName("Heartbeat Protocol Tests")
+class HeartbeatProtocolTests {
```

Comment:
It seems silly because it is so simple, but there should be tests to verify that `shouldInitiate` and `shouldAccept` return `true`. If someone ever changes that behavior, a test should fail. Plus, it's easy coverage to get

Reply:
behavior of `HeartbeatProtocol::shouldInitiate` should already be fully covered in the `shouldInitiate` test, but I'll add a test for `shouldAccept` as well 👍